src/CommandingOfficers/AW2/AW2Commander.java

```diff
@@ -24,19 +24,15 @@ public AW2Commander(CommanderInfo info, GameScenario.GameRules rules)
   @Override
   public int calculateCombatCharge(UnitDelta minion, UnitDelta enemy, boolean isCounter)
   {
-    if( null != getActiveAbility() )
-      return 0;
     if( minion == null || enemy == null )
       return 0;
 
-    int guiHPLoss  = minion.getHealthDamage() / 10;
-    int guiHPDealt =  enemy.getHealthDamage() / 10;
+    int guiHPLoss  = minion.getHPDamage();
+    int guiHPDealt =  enemy.getHPDamage();
 
     int power = 0; // value in funds of the charge we're getting
 
-    // Add up the funds value of the damage done to both participants.
     power += guiHPLoss * minion.unit.getCost() / 10;
-    // The damage we deal is worth half as much as the damage we take, to help powers be a comeback mechanic.
     power += guiHPDealt * enemy.unit.getCost() / 10 / 2;
```

Comment:
Here and elsewhere you took away the comments but left the code. Why?

Reply:
The design intent isn't ours.

Reply:
Might have to unpack that for me. The comments explain what the code is doing, right? Why remove that?

Reply:
The first comment isn't any clearer than the code itself. It's clutter.

The second comment's first clause mentions "half", which is cool I guess.
The second comment's conclusion is about design intent, and the design intent isn't really relevant since we're just matching cart behavior there.

I guess I could put in "Charge based on funds damage taken + half dealt (to match the cart)", but:
That's still just kind of duplicating the code.
The whole point of the class is to match cart logic, so mentioning that is redundant.

Reply:
I'd probably agree that it's irrelevant to mention that we are mimicking cart behavior (though not because that's the purpose of the class, since the class doesn't state that elsewhere either). Whether the design intent is ours is also irrelevant; comments should communicate the intent regardless of whose intent it initially was.

I do agree that pairing lines of code with equivalent comments is a bit silly, but I like having a comment to explain a block of code in plain English since 1) is easier to parse, 2) knowing intent helps to parse the code too, and 3) it also provides confirmation of correctness if they match, and an immediate red flag if they don't. Without the intent you just have to trust the code is right.

But this code also isn't that complex; it's pretty clear as is, aside from the magic numbers (though the part about comeback mechanics in the game's design doesn't come through, but hey). I'd still leave a comment (and purge the magic, ofc), but up to you I s'pose.

Reply:
> I'd probably agree that it's irrelevant to mention that we are mimicking cart behavior (though not because that's the purpose of the class, since the class doesn't state that elsewhere either).

It's in the name: `AW2Commander`
It'd feel pretty redundant to slap on a "To implement AW2 COs" docstring on there.
> Whether the design intent is ours is also irrelevant; comments should communicate the intent regardless of whose intent it initially was.

Suuuure, but we don't actually _know_ what the original design intent was.
We can make assumptions, but that's what they'd be.
The intent of *this* code is to... do whatever the cart code does.
> I do agree that pairing lines of code with equivalent comments is a bit silly, but I like having a comment to explain a block of code in plain English since 1) is easier to parse, 2) knowing intent helps to parse the code too, and 3) it also provides confirmation of correctness if they match, and an immediate red flag if they don't. Without the intent you just have to trust the code is right.
> 
> But this code also isn't that complex; it's pretty clear as is, aside from the magic numbers (though the part about comeback mechanics in the game's design doesn't come through, but hey). I'd still leave a comment (and purge the magic, ofc), but up to you I s'pose.

Fair.
I'm not sure killing the magic in this instance really helps much, but "document the intended behavior" seems pretty defensible.